pkg/collector/corechecks/servicediscovery/usm/erlang.go

```diff
@@ -0,0 +1,82 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package usm
+
+import (
+	"path/filepath"
+	"strings"
+)
+
+type erlangDetector struct {
+	ctx DetectionContext
+}
+
+func newErlangDetector(ctx DetectionContext) detector {
+	return &erlangDetector{ctx: ctx}
+}
+
+func (e erlangDetector) detect(args []string) (ServiceMetadata, bool) {
+	name := detectErlangAppName(args)
+	if name != "" && name != "beam" {
+		return NewServiceMetadata(name, CommandLine), true
+	}
+	return ServiceMetadata{}, false
+}
+
+func detectErlangAppName(cmdline []string) string {
+	var progname string
+	var home string
+
+	// Parse command line looking for -progname and -home flags.
+	// We support both separated (-progname erl) and concatenated (-prognameerl) forms.
```

Comment:
Looks odd that Erlang would support a command line option like this `-prognameerl` (how would it distinguish options with the same prefix?). Have you verified that this is the case?